docs/userguide/datafiles.rst

```diff
@@ -7,86 +7,223 @@ are placed in a platform-specific location.  However, the most common use case
 for data files distributed with a package is for use *by* the package, usually
 by including the data files **inside the package directory**.
 
+include_package_data
+====================
+
 Setuptools offers three ways to specify this most common type of data files to
-be included in your package's [#datafiles]_.
-First, you can simply use the ``include_package_data`` keyword, e.g.::
+be included in your packages.
+First, you can simply use the ``include_package_data`` keyword.
+For example, if the package tree looks like this::
+
+    project_root_directory
+    ├── setup.py        # and/or setup.cfg, pyproject.toml
+    └── src
+        └── mypkg
+            ├── __init__.py
+            ├── data1.rst
+            ├── data2.rst
+            ├── data1.txt
+            └── data2.txt
+
+and you supply this configuration:
+
+.. tab:: setup.cfg
+
+   .. code-block:: ini
+
+        [options]
+        # ...
+        packages = find:
+        package_dir =
+            = src
+        include_package_data = True
+
+        [options.packages.find]
+        where = src
+
+.. tab:: setup.py
+
+   .. code-block:: python
 
     from setuptools import setup, find_packages
     setup(
-        ...
+        # ...,
+        packages=find_packages(where="src"),
+        package_dir={"": "src"},
         include_package_data=True
     )
 
-This tells setuptools to install any data files it finds in your packages.
-The data files must be specified via the |MANIFEST.in|_ file.
-(They can also be tracked by a revision control system, using an appropriate
-plugin such as :pypi:`setuptools-scm` or :pypi:`setuptools-svn`.
-See the section below on :ref:`Adding Support for Revision
-Control Systems` for information on how to write such plugins.)
+.. tab:: pyproject.toml (**EXPERIMENTAL**) [#experimental]_
 
-If you want finer-grained control over what files are included (for example,
-if you have documentation files in your package directories and want to exclude
-them from installation), then you can also use the ``package_data`` keyword,
-e.g.::
+   .. code-block:: toml
 
-    from setuptools import setup, find_packages
-    setup(
-        ...
-        package_data={
-            # If any package contains *.txt or *.rst files, include them:
-            "": ["*.txt", "*.rst"],
-            # And include any *.msg files found in the "hello" package, too:
-            "hello": ["*.msg"],
-        }
-    )
+        [tool.setuptools]
+        # ...
+        # By default, include-package-data is true in pyproject.toml, so you do
+        # NOT have to specify this line.
+        include-package-data = true
+
+        [tool.setuptools.packages.find]
+        where = ["src"]
```

Comment:
The behaviour of `pyproject.toml` (if no `py-modules` or `packages configuration is given) is by default the same as the following (so in theory it is not explicitly required):

```toml
[tool.setuptools.packages.find]
where = ["src"]
```

Reply:
This was for the same reason as the other similar snippet. Your call on this - I can remove this if it is not required.